core/src/main/scala/com/typesafe/tools/mima/lib/analyze/method/MethodChecker.scala

```diff
@@ -35,9 +43,9 @@ private[analyze] class ClassMethodChecker extends BaseMethodChecker {
     if (method.nonAccessible)
       None
     else if (method.isDeferred)
-      super.check(method, inclazz.lookupMethods(method.bytecodeName))
+      super.check(method, inclazz, _.lookupMethods(method.bytecodeName))
     else
-      super.check(method, inclazz.lookupClassMethods(method.bytecodeName))
+      super.check(method, inclazz, _.lookupClassMethods(method.bytecodeName))
```

Comment:
The lookup function is used for old and new, I wonder if there is a case when we get a misleading error message if a method is removed and only exists in a parent trait in the new version.

Generally, maybe can you explain why a different lookup function is used if `meth.isDeferred`?

Reply:
No, I can't.  It's a part of the code that I don't understand and just try to preserve the existing semantics.

Reply:
👍 for your pragmatism